pkg/kube/utils.go

```diff
@@ -171,3 +174,20 @@ func PVCContainsReadOnlyAccessMode(pvc *corev1.PersistentVolumeClaim) bool {
 
 	return false
 }
+
+// AddDebugLabelsToPodOptions adds additional label selector to `PodOptions`,
+// provided the context has a key starting with `keyPrefix`.
+func AddDebugLabelsToPodOptions(ctx context.Context, options *PodOptions, keyPrefix, keySuffix string) {
+	fields := field.FromContext(ctx)
+	if fields != nil {
+		for _, f := range fields.Fields() {
+			if strings.HasPrefix(f.Key(), keyPrefix) {
+				value := f.Value().(string)
+				if options.Labels == nil {
+					options.Labels = make(map[string]string)
+				}
+				options.Labels[keyPrefix+keySuffix] = value
+			}
+		}
+	}
```

Comment:
```suggestion
	if fields == nil {
	        return
	}
	for _, f := range fields.Fields() {
		if !strings.HasPrefix(f.Key(), keyPrefix) {
		        continue
		}
		value := f.Value().(string)
		if options.Labels == nil {
			options.Labels = make(map[string]string)
		}
		options.Labels[labelKey] = value
	}
```
Its best practice to reduce nesting in Go - https://github.com/uber-go/guide/blob/master/style.md#reduce-nesting

Reply:
Done